packages/compass-indexes/src/components/create-index-form/create-index-form.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { Component } from 'react';
+import {
+  css,
+  withTheme,
+  Label,
+  spacing,
+  Accordion,
+} from '@mongodb-js/compass-components';
+
+import CreateIndexFields from '../create-index-fields';
+import { hasColumnstoreIndexesSupport } from '../../utils/has-columnstore-indexes-support';
+import UniqueIndexCheckbox from './unique-index';
+import TTLCollapsibleFieldSet from './ttl';
+import PartialFilterCollapsibleFieldSet from './partial-filter-expression';
+import CustomCollationCollapsibleFieldSet from './custom-collation';
+import WildcardProjectionCollapsibleFieldSet from './wildcard-projection';
+import ColumnstoreProjectionCollapsibleFieldSet from './columnstore-projection';
+import IndexNameCollapsibleFieldSet from './index-name';
+
+const createIndexModalFieldsStyles = css({
+  margin: `${spacing[4]}px 0 ${spacing[5]}px 0`,
+});
+
+const createIndexModalOptionStyles = css({
+  paddingLeft: spacing[1] + 2,
+});
+
+type IndexField = { name: string; type: string };
+
+export type CreateIndexProps = {
+  darkMode?: boolean;
+  error?: string;
+  inProgress: boolean;
+  fields: IndexField[];
+  newIndexField?: string;
+  schemaFields: string[];
+
+  isVisible: boolean;
+  isUnique: boolean;
+
+  useIndexName: boolean;
+  useTtl: boolean;
+  usePartialFilterExpression: boolean;
+  useCustomCollation: boolean;
+  useWildcardProjection: boolean;
+  useColumnstoreProjection: boolean;
+
+  name: string;
+  ttl?: string;
+  partialFilterExpression?: string;
+  wildcardProjection: string;
+  collationString?: string;
+  columnstoreProjection: string;
+
+  namespace: string;
+  serverVersion: string;
+
+  // TODO: Refactor modules to get rid of return any.
```

Comment:
Can these be `void` instead of `any`? Maybe I'm missing something but it looks like they aren't expected to return any values.